scala-thoughts.md

```diff
@@ -0,0 +1,37 @@
+I really like how functional programming is done in Scala.
+I don't have extensive experience using languages for functional programming,
+so I don't have the best sense of scale for this, but Scala gives the 
+programmer a lot of built-in functions to work with.
+As an example, I thought the `Rainfall.scala` exercise would be difficult to 
+write using functional programming.
+But reading Prof. Ben's solution, I was surprised that there is a very
+convenient function for `Rainfall.scala`-type problems.
+In retrospect, there are definitely many scenarios that could use `takeWhile`.
+I would like to learn more Scala functions by reading others' code as opposed
+to trawling through documentation.
```

Comment:
Definitely agree. Scala gives the programmer so much flexibility that it's really hard to get a sense of the "Scala-y" way of doing things, without seeing how other people use that flexibility.